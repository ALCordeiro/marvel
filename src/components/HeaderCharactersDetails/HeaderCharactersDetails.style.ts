import styled from 'styled-components';

export const HeaderContainer = styled.div``;

export const LogoContainer = styled.div`
    display: flex;
    max-width: 330px;
    max-height: 120px;
    justify-content: center;
    padding-bottom: 20px;
`;
