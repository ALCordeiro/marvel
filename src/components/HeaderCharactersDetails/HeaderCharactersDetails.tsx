import React from 'react';
import { Link } from 'react-router-dom';

import {
    HeaderContainer,
    LogoContainer,
} from './HeaderCharactersDetails.style';

export const HeaderCharactersDetailsComponent: React.FC = () => (
    <HeaderContainer>
        <LogoContainer>
            <Link to={{ pathname: '/' }}>
                <img
                    src={`${window.location.origin}/assets/marvel-logo.png`}
                    alt="marvel-logo"
                />
            </Link>
        </LogoContainer>
    </HeaderContainer>
);

export default HeaderCharactersDetailsComponent;
