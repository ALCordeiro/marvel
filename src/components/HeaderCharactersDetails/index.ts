export { default as HeaderCharactersDetailsComponent } from './HeaderCharactersDetails';
