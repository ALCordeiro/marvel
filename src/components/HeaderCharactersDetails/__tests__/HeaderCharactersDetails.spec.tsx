/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { HeaderCharactersDetailsComponent } from '../HeaderCharactersDetails';

let wrapper;

beforeEach(() => {
    wrapper = render(
        <Router>
            <HeaderCharactersDetailsComponent />
        </Router>,
    );
});

test('should render', async () => {
    const { container } = wrapper;
    expect(container.firstChild);
});

test('should match snapshot', () => {
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
});

test('should render with a link tag', () => {
    const { container } = wrapper;
    expect(container.firstChild.firstChild.firstChild.nodeName).toBe('A');
    expect(document.querySelector('a').getAttribute('href')).toBe('/');
});

test('should render with a IMG tag', () => {
    const { container } = wrapper;
    expect(container.firstChild.firstChild.firstChild.firstChild.nodeName).toBe(
        'IMG',
    );
    expect(document.querySelector('img').getAttribute('alt')).toBe(
        'marvel-logo',
    );
});
