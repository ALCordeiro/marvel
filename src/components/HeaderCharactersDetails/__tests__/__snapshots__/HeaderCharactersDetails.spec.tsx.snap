// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-bdnxRM"
>
  <div
    class="sc-gtsrHT bsaVBs"
  >
    <a
      href="/"
    >
      <img
        alt="marvel-logo"
        src="http://localhost/assets/marvel-logo.png"
      />
    </a>
  </div>
</div>
`;
